core/src/main/java/com/linecorp/armeria/internal/common/metric/RequestMetricSupport.java

```diff
@@ -343,6 +353,22 @@ public Counter writeTimeouts() {
         public Counter responseTimeouts() {
             return responseTimeouts;
         }
+
+        @Override
+        public DistributionSummary retries() {
+            final DistributionSummary retries = this.retries;
+            if (retries != null) {
+                return retries;
+            }
+
+            final DistributionSummary distributionSummary = parent.summary(
+                    idPrefix.name("retries"), idPrefix.tags());
+            if (retriesUpdater.compareAndSet(this, null, distributionSummary)) {
+                return distributionSummary;
+            }
```

Comment:
That's a good suggestion. 👍

Reply:
@ikhoon 
Thanks a lot for your suggestion. I think I need to use `childrenSize` to detect whether `RetryingClient` is used. I found that the constructor used in `MicrometerUtil.register` method and it requires a `BiFunction` constructor while the new constructor have 3 parameters. I'm not sure whether we can define a new `register` method which accept more than 2 parameters.
```
final ClientRequestMetrics metrics = MicrometerUtil.register(registry, idPrefix,
                                                                     ClientRequestMetrics.class,
                                                                     DefaultClientRequestMetrics::new);
```

Reply:
I guess if we want to follow this idea, we may define a separate class i.e. `ExtendedDefaultClientRequestMetrics` which extends `DefaultClientRequestMetrics`.

Having said this, I'm honestly not sure if there is much merit to precomputing each counter/gauge.

What do you think of just registering the distributionSummary on the fly @ikhoon @minwoox ?
I don't think there's much benefit to caching the distributionSummary since it will be invoked once anyways.

e.g.
```
        @Override
        public DistributionSummary successAttempts() {
            return parent.summary(idPrefix.name("actual.requests.attempts"),
                                  idPrefix.tags("result", "success"));
        }
```

Reply:
What I understood from @ikhoon's comment was to check if there's a `RetryingClient` in `AbstractMetricCollectingClient` and passing it:
```java
abstract class AbstractMetricCollectingClient<I extends Request, O extends Response>
        extends SimpleDecoratingClient<I, O> {

    ...
    @Override
    public final O execute(ClientRequestContext ctx, I req) throws Exception {
        final boolean addAttemptMetrics = as(AbstractRetryingClient.class) != null ||
                                            as(RedirectingClient.class) != null; // RedirectingClient should moved to internal package to make it public.
        RequestMetricSupport.setup(ctx, REQUEST_METRICS_SET, meterIdPrefixFunction, false,
                                   successFunction != null ? successFunction::test
                                                           : ctx.options().successFunction(), addAttemptMetrics);
        return unwrap().execute(ctx, req);
    }
}
```

Reply:
`DefaultClientRequestMetrics` is an internal class so both are fine to me.
I'm slightly leaning toward keeping one class to keep it simple though. 😄

Reply:
@nhacvuong gentle ping. 😉

Reply:
@minwoox 
I updated code base on jrhee17 's suggestion. Please kindly help me take a look.

Reply:
@nhacvuong Could you use this approach?
https://github.com/line/armeria/pull/4707#discussion_r1145804280
Then we can create the `actualRequests` if `retryable` is true.

Reply:
Sorry about the misdirection. 😓 
Had a chat with @ikhoon and @jrhee17, realized that we don't need to use `retryable`. Let me continue to review on this PR. Thanks!